### DISCUSSION TITLE
230718 2차 데모데이 발표 회의 (10:30 ~ 11:30)

### DISCUSSION THREAD

**Discussion by Cyma-s** [woowacourse-teams/2023-shook#62] [회의]
## 회의 안건
- 2차 데모데이 발표자 선정
- 2차 데모데이 발표 내용 선정

## 예상 시간
- 1시간

**Comment by seokhwan-an** [comment]
# 2차 데모데이 발표

### 목차

- 기술 스택 정의
- git branch 전략 재확립
    - 왜 github flow를 채택했는가?
        - `운영 서비스가 없는 상태`에서 개발만 하는 와중에 main과 develop을 분리해서 관리하는 것이 불편했다. → 추후에 운영이 생기면 QA용도로 develop 부활할지도..
- 2차 요구사항 (간단히 한마디)
    - CI
        - git actions를 쓴 이유?
    - CD
        - ec2 내부에 runner를 실행시켜 실시간 배포를 수행한다.
- 핵심 기능 발표
    - 페르소나 재정의
    - 우리가 구현한 핵심 기능 현황
        - 파트 등록하는 기능
            - 본인이 등록한 파트의 순위를 나타내는 기능
        - 공유하는 기능
            - youtube 링크에 시작 초와 끝 초를 붙여 본인이 등록한 부분만 공유할 수 있게 한다.
- 핵심 기능 데모
    - 등록하는 거랑
    - 간편하게 공유하는 모습

**Reply by splitCoding** [reply]
## 1. 기술 스택 정의 (아코)
- 기술 스택 로고 목록

## 2. Git Branch 전략 재확립 (베로)
- 이전, 이후 시각화 자료
- 이유
    - 현재 프로젝트 상황에서는 단순한 Github Flow가 관리가 더 용이하다.
        - develop, main 의 싱크를 신경쓸 필요 없다.
        - 브랜치가 파생될 수 있는 곳이 main 밖에 없어 충돌 가능성이 적다.

## 3. CI 자동화 (스플릿)
- CI에 git actions를 쓴 이유 ( 지금 상황에서 불편함이 없다. )

## 4. 페르소나 재정의 (도밥)
- 재정의한 페르소나
- 재정의한 이유
- 재정의한 기준

## 5. 핵심 기능 발표 (코난)
- 파트 등록 기능
    - 등록한 파트의 순위를 알려준다.
- 공유 기능
    - 실행할 수 있는 유투브 링크를 제공한다.

**Comment by Cyma-s** [comment]
## 2차 데모데이 발표자
- BE: 바론
- FE: 우코

**Comment by splitCoding** [comment]
## 데모 시연 방식
- 로컬 프론트 서버에서 EC2 백엔드 서버로 요청하며 시연한다.
- 순서 
    1. 파트 등록 과정 ( 반복 : 동일 구간 등록 2번, 다른 구간 등록 1번 )
        1. 파트 등록 페이지에서 등록한다.
        2. 등록 후 나오는 모달 창에서 순위를 확인한다.
        3. 모달 창에서 공유하기를 클릭하면 해당 파트를 실행할 수 있는 URL 이 생성된다.
    2. 해당 URL에 접속하여 실행
